integrations/Criteo/browser.js

```diff
@@ -0,0 +1,222 @@
+    logger.debug("in Criteo isLoaded");
+    return !!(window.criteo_q && window.criteo_q.push !== Array.prototype.push);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  isReady() {
+    logger.debug("in Criteo isReady");
+    return !!(window.criteo_q && window.criteo_q.push !== Array.prototype.push);
+  }
+
+  handleCommonFields(rudderElement) {
+    const { message } = rudderElement;
+    const { properties } = message;
+
+    const setEmail = {};
+    const setZipcode = {};
+
+    const finalRequest = [
+      { event: "setCustomerId", id: md5(message.userId) },
+      { event: "setRetailerVisitorId", id: md5(message.anonymousId) },
+    ];
+
+    if (properties.email) {
+      const email = properties.email.trim().toLowerCase();
+      setEmail.event = "setEmail";
+      setEmail.hash_method = this.hashMethod;
+      setEmail.email = this.hashMethod === "md5" ? md5(email) : email;
+      finalRequest.push(setEmail);
+    }
+
+    if (properties.zipCode) {
+      setZipcode.event = "setZipcode";
+      setZipcode.zipCode = properties.zipCode || properties.zip;
+      finalRequest.push(setZipcode);
+    }
+
+    return finalRequest;
+  }
+
+  page(rudderElement) {
+    const { name, properties } = rudderElement.message;
+
+    const finalPayload = this.handleCommonFields(rudderElement);
+
+    if (
+      name === "home" ||
+      (this.url && this.url === window.location.href) ||
+      properties.url === this.url
+    ) {
+      const homeEvent = {
+        event: "viewHome",
+      };
+      finalPayload.push(homeEvent);
+    }
+
+    window.criteo_q.push(finalPayload);
+  }
+
+  track(rudderElement) {
+    const { event, properties } = rudderElement.message;
+
+    const finalPayload = this.handleCommonFields(rudderElement);
+
+    // Product tag
+    if (event === "Product Viewed") {
+      const viewItemObject = {
+        event: "viewItem",
+        item: String(properties.product_id),
+      };
+
+      if (properties.price && !Number.isNaN(parseFloat(properties.price))) {
+        viewItemObject.price = parseFloat(properties.price);
+      }
+
+      if (properties.availability === 1 || properties.availability === 0) {
+        viewItemObject.availability = properties.availability;
+      }
+      if (!viewItemObject.item) {
+        // productId is mandatory
+        return;
+      }
+      finalPayload.push(viewItemObject);
+    }
+
+    // Basket/cart tag && sales tag
+    if (event === "Cart Viewed" || event === "Order Completed") {
+      const productInfo = [];
+      let elementaryProduct;
+      properties.products.forEach((product) => {
+        elementaryProduct = {
+          id: String(product.product_id),
+          price: parseFloat(product.price),
+          quantity: parseInt(product.quantity, 10),
+        };
+
+        if (
+          elementaryProduct.id &&
+          elementaryProduct.price &&
+          !Number.isNaN(parseFloat(elementaryProduct.price)) &&
+          elementaryProduct.quantity &&
```

Comment:
`productInfo` is initialized as an empty array. where is the `id` coming from?

Reply:
no, because according to criteo docs, product_id is named as id for sales and basket tag